crates/provider/src/fillers/mod.rs

```diff
@@ -329,7 +339,9 @@ where
     }
 
     fn call<'req>(&self, tx: &'req N::TransactionRequest) -> EthCall<'req, N, Bytes> {
-        self.inner.call(tx)
+        let prepare_res = self.prepare_call(tx.clone());
```

Comment:
this seems okay, here but the alternative is moving this inside the ethcallfiller and use the async call there.

because isn't this now just the same as calling self.inner.call?

actually perhaps it's easier that way for now, so let's simplify this drastically and only add this to relevant calls and not introduce the ethcallfiller type.

but the proper solution here would be to move this into the future I think